lib/spack/spack/package.py

```diff
@@ -1543,7 +1559,11 @@ def do_install(self, **kwargs):
                     dep.package.bootstrap_compiler(**comp_kwargs)
                 dep.package.do_install(**dep_kwargs)
 
-        # Then, install the package proper
+        # There's nothing left to do for a bundle (or no-code) package
+        if not self.has_code:
+            return
```

Comment:
Doesn't this avoid sticking the no-source package in the database entirely?  I think we should allow `install()` to run for no-source packages, and generate the DB entry for them; just don't fetch.

@becker33: thoughts?

Reply:
Ah.  Greg just told me there's a "bundle" package that uses `install` to manage metadata about the installation.